overlord/fdestate/fdemgr.go

```diff
@@ -49,12 +50,31 @@ func Manager(st *state.State, runner *state.TaskRunner) *FDEManager {
 	return m
 }
 
+// Ensure implements StateManager.Ensure
 func (m *FDEManager) Ensure() error {
 	return nil
 }
 
-func (m *FDEManager) resealKeyForBootChains(method device.SealingMethod, rootdir string, params *boot.ResealKeyForBootChainsParams, expectReseal bool) error {
-	return backend.ResealKeyForBootChains(method, rootdir, params, expectReseal)
+// StartUp implements StateStarterUp.Startup
+func (m *FDEManager) StartUp() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+	return initializeState(m.state)
+}
+
+func (m *FDEManager) resealKeyForBootChains(unlocker boot.Unlocker, method device.SealingMethod, rootdir string, params *boot.ResealKeyForBootChainsParams, expectReseal bool) error {
+	doUpdate := func(role string, containerRole string, bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte) error {
+		if unlocker != nil {
+			m.state.Lock()
+			defer m.state.Unlock()
+		}
+		return updateParameters(m.state, role, containerRole, bootModes, models, tpmPCRProfile)
+	}
+	if unlocker != nil {
+		locker := unlocker()
+		defer locker()
+	}
+	return backend.ResealKeyForBootChains(doUpdate, method, rootdir, params, expectReseal)
 }
```

Comment:
This function is not covered by tests in this package, but still covered by tests from `overlord/managers_test.go`. I wonder if we should add tests in the current package. Or move them.

Reply:
Done. It was easier than expected.